Tella/Scenes/Audio/AudioPlayer/ViewModel/EditAudioViewModel.swift

```diff
@@ -62,99 +35,43 @@ class EditAudioViewModel: ObservableObject {
             self.timeDuration = self.audioPlayerManager.audioPlayer.duration
             self.generateTimeLabels()
             self.endTime = self.timeDuration
-
+            
         }
     }
-
+    
     private func listenToAudioPlayerUpdates() {
         self.audioPlayerManager.audioPlayer.currentTime.sink { value in
             self.currentTime = value.formattedAsHHMMSS()
             self.updateOffset(time: Double(value) )
-        }.store(in: &self.cancellable)
-
+        }.store(in: &self.cancellables)
+        
         self.audioPlayerManager.audioPlayer.audioPlayerDidFinishPlaying.sink { [self] value in
             isPlaying = false
-        }.store(in: &self.cancellable)
+        }.store(in: &self.cancellables)
     }
-    
-    func onAppear() {
-        guard let fileExtension = file?.fileExtension else { return }
-        let url = appModel.vaultManager.saveDataToTempFile(data: currentData, pathExtension: fileExtension)
-        guard let url else { return }
-        
-        self.audioUrl = url
-    }
-    
-    func onDisappear() {
-        self.onPause()
-    }
-    
-    func isDurationHasChanged() -> Bool {
-        return self.endTime != self.timeDuration || self.startTime != 0.0
-    }
-    
-    func trimAudio() {
+
+    override func trim() {
         Task {
             do {
-                guard let trimmedAudioUrl = try await audioUrl?.trimAudio(newName: "\(newFileName).m4a", startTime: startTime, endTime: endTime) else { return }
+                let copyName: String = file?.getCopyName(from: appModel.vaultFilesManager) ?? ""
+                guard let trimmedAudioUrl = try await fileURL?.trimMedia(newName: "\(copyName).m4a", startTime: startTime, endTime: endTime, type: .mp4) else { return }
```

Comment:
What do you think about passing only the `newName` as a parameter and determining the extension from the URL inside the  `trimMedia` function ?